node/popn.go

```diff
@@ -760,12 +760,19 @@ func (nd *node) get(ctx context.Context, c cid.Cid, args *GetArgs) error {
 				return err
 			}
 		}
+
 		// Register new blocks in our supply by default
-		err = nd.exch.Index().SetRef(&exchange.DataRef{
+		ref := &exchange.DataRef{
```

Comment:
If we're fetching content from an existing ref we should try getting it first:
```suggestion
                 ref, err := nd.exch.Index().GetRef(c)
                 if err != nil {
                 // Create a new ref
		ref = &exchange.DataRef{
		 ...
		 }
		 }
```